oracle-plugin/src/e2e-test/java/io.cdap.plugin/OracleClient.java

```diff
@@ -52,6 +62,108 @@ public static int countRecord(String table, String schema) throws SQLException,
     }
   }
 
+  /**
+   * Extracts entire data from source and target tables.
+   * @param sourceTable table at the source side
+   * @param targetTable table at the sink side
+   * @return number of different values in source and target side
+   */
+  public static boolean validateRecordValues(String schema, String sourceTable, String targetTable)
+    throws SQLException, ClassNotFoundException {
+    String getSourceQuery = "SELECT * FROM " + schema + "." + sourceTable;
+    String getTargetQuery = "SELECT * FROM " + schema + "." + targetTable;
+    try (Connection connect = getOracleConnection()) {
+      connect.setHoldability(ResultSet.HOLD_CURSORS_OVER_COMMIT);
+      Statement statement1 = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE,
+                                                     ResultSet.HOLD_CURSORS_OVER_COMMIT);
+      Statement statement2 = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE,
+                                                     ResultSet.HOLD_CURSORS_OVER_COMMIT);
+      ResultSet rsSource = statement1.executeQuery(getSourceQuery);
+      ResultSet rsTarget = statement2.executeQuery(getTargetQuery);
+      return resultSetDataCompare(rsSource, rsTarget);
+    }
+  }
+
+  /**
+   * Counts the number of columns with different data in source table and sink table in countDifferentValues variable.
+   * @param rsSource result set of the source table data
+   * @param rsTarget result set of the target table data
+   * @return countDifferentValues
+   */
+  public static boolean resultSetDataCompare(ResultSet rsSource, ResultSet rsTarget) throws SQLException {
+    boolean differentValues = true;
+    ResultSetMetaData mdSource = rsSource.getMetaData();
+    ResultSetMetaData mdTarget = rsTarget.getMetaData();
+    int columns = mdSource.getColumnCount();
+    while (rsSource.next() && rsTarget.next()) {
+      int count = 1;
+      while (count <= columns) {
+        differentValues = false;
+        String columnTypeName = mdSource.getColumnTypeName(count);
+        int columnType = mdSource.getColumnType(count);
+        String columnName = mdSource.getColumnName(count);
+        switch (columnType) {
+          case OracleSourceSchemaReader.BFILE:
```

Comment:
BFILE is not written in the Oracle Sink, please remove this check.

Reply:
Removed this